src/pages/home/report/ReportActionItem.js

```diff
@@ -345,20 +347,40 @@ function ReportActionItem(props) {
             );
         } else if (props.action.actionName === CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENTQUEUED) {
             const submitterDisplayName = PersonalDetailsUtils.getDisplayNameOrDefault(props.personalDetailsList, [props.report.ownerAccountID, 'displayName'], props.report.ownerEmail);
+            const paymentType = lodashGet(props.action, 'originalMessage.paymentType', '');
             const shouldShowAddCreditBankAccountButton =
-                ReportUtils.isCurrentUserSubmitter(props.report.reportID) && !store.hasCreditBankAccount() && !ReportUtils.isSettled(props.report.reportID);
-
+                ReportUtils.isCurrentUserSubmitter(props.report.reportID) &&
+                !store.hasCreditBankAccount() &&
+                !ReportUtils.isSettled(props.report.reportID) &&
+                paymentType !== CONST.IOU.PAYMENT_TYPE.EXPENSIFY;
+            const shouldShowEnableWalletButton =
+                ReportUtils.isCurrentUserSubmitter(props.report.reportID) && paymentType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY && !ReportUtils.isSettled(props.report.reportID);
             children = (
-                <ReportActionItemBasicMessage message={props.translate('iou.waitingOnBankAccount', {submitterDisplayName})}>
-                    {shouldShowAddCreditBankAccountButton ? (
+                <ReportActionItemBasicMessage
+                    message={
+                        shouldShowEnableWalletButton
+                            ? props.translate('iou.waitingOnEnableWallet', {submitterDisplayName})
+                            : props.translate('iou.waitingOnBankAccount', {submitterDisplayName})
+                    }
+                >
+                    {shouldShowAddCreditBankAccountButton && (
                         <Button
                             success
                             style={[styles.w100, styles.requestPreviewBox]}
                             text={props.translate('bankAccount.addBankAccount')}
                             onPress={() => BankAccounts.openPersonalBankAccountSetupView(props.report.reportID)}
                             pressOnEnter
                         />
-                    ) : null}
+                    )}
+                    {shouldShowEnableWalletButton && (
+                        <Button
+                            success
+                            style={[styles.w100, styles.requestPreviewBox]}
+                            text={props.translate('iou.enableWallet')}
+                            onPress={() => Navigation.navigate(ROUTES.ENABLE_PAYMENTS)}
```

Comment:
If the user does not have a personal bank account set up, we should redirect them to the Personal Bank account page first.

Can we use the [KYCWall](https://github.com/Expensify/App/blob/a24e6a09b86f703f8ccf2e8f4bb29a4aabd6c761/src/components/SettlementButton.js#L201-L224) here?